ccloader/js/mod.js

```diff
@@ -175,13 +171,20 @@ export class Mod {
 	}
 
 	async _loadManifest() {
-		const file = this.file;
+		let file;
 		let data;
 		try {
+			file = this.baseDirectory + 'ccmod.json'
 			data = await this.filemanager.getResourceAsync(file);
-		} catch (e) {
-			console.error(e);
-			return;
+		} catch (e1) {
+			try {
+				file = this.baseDirectory + 'package.json'
+				data = await this.filemanager.getResourceAsync(file);
+			} catch (_e2) {
+				console.error(e1);
+				console.error(e2);
+				return;
+			}
 		}
```

Comment:
If `file` was given to the constructor, this ugly try/catch logic would be unnecessary.

Reply:
See my other comment. Then those two `try`/`catch`es will be moved to packed mod detection code.